mlflow/utils/validation.py

```diff
@@ -397,7 +397,7 @@ def _validate_model_version(model_version):
         )
 
 
-def _validate_model_alias_name(model_alias_name):
+def _validate_model_alias_name(model_alias_name, perform_latest_check=True):
```

Comment:
We can probably get rid of this boolean / `latest`-specific logic if we switch to removing validation on reads + enforcing it on writes. And then this PR can be about removing validation on fetching model versions by alias